File: 3151-minimum-processing-time/minimum-processing-time.py
class Solution:
    def minProcessingTime(self, processorTime: List[int], tasks: List[int]) -> int:
        processorTime.sort()
        tasks.sort(reverse=True)
        print(tasks)
        mintime = k = 0
        for i in range(len(processorTime)):
            time = processorTime[i] + tasks[k]
            mintime = max(mintime, time)
            k += 4

        return mintime

File: 3151-minimum-processing-time/minimum-processing-time.cpp
class Solution {
public:
    int minProcessingTime(vector<int>& processorTime, vector<int>& tasks) {
        sort(processorTime.begin(), processorTime.end());
        sort(tasks.rbegin(), tasks.rend());
        int mintime = 0, k = 0;
        for (int i = 0; i < processorTime.size(); i++)
        {
            int time = processorTime[i] + tasks[k];
            mintime = max(mintime, time);
            k += 4;
        }
        return mintime;
    }
};